src/app/wallet-activation/user-activation/user-activation.component.ts

```diff
@@ -17,16 +18,21 @@ import swal from 'sweetalert2';
 export class UserActivationComponent implements OnInit {
     users: CooperativeUser[];
 
-    constructor(private activationService: WalletCooperativeWalletService, private broadService: BroadcastService) {
+    constructor(private activationService: WalletCooperativeWalletService,
+                private broadService: BroadcastService) {
     }
 
     ngOnInit() {
-        SpinnerUtil.showSpinner();
+        this.fetchWalletToActivate();
+    }
 
-        this.activationService.getUnactivatedUserWallets().subscribe((res) => {
-            this.users = res.users;
-            SpinnerUtil.hideSpinner();
-        }, hideSpinnerAndDisplayError);
+    fetchWalletToActivate() {
+        SpinnerUtil.showSpinner();
+        this.activationService.getUnactivatedUserWallets()
+            .subscribe((res) => {
+                this.users = res.users;
+                SpinnerUtil.hideSpinner();
```

Comment:
as `hideSpinnerAndDisplayError` will be called in the final block, you can remove  `hideSpinner()`.

```suggestion
```

Reply:
djakldkssa